tests/sklearn/test_fhe_training.py

```diff
@@ -293,259 +302,236 @@ def test_clear_fit(
+    # Make sure weight, bias and prediction values are identical between clear and partial fitting
     assert array_allclose_and_same_shape(weights_disable, weights_partial)
     assert array_allclose_and_same_shape(bias_disable, bias_partial)
-    assert array_allclose_and_same_shape(weights_partial_2, weights_partial)
-    assert array_allclose_and_same_shape(bias_partial_2, bias_partial)
     assert array_allclose_and_same_shape(y_pred_proba_disable, y_pred_proba_partial)
     assert array_allclose_and_same_shape(y_pred_class_disable, y_pred_class_partial)
 
-    if not fit_intercept:
-        assert bias_partial == numpy.zeros((1, 1))
-        assert bias_disable == numpy.zeros((1, 1))
-        assert bias_simulate == numpy.zeros((1, 1))
-
-    # Initialize a model for fitting
-    with warnings.catch_warnings():
-        # FHE training is an experimental feature and a warning is raised each time `fit_encrypted`
-        # is set to True
-        warnings.filterwarnings("ignore", message="FHE training is an experimental feature.*")
-
-        model_from_another = SGDClassifier(
-            n_bits=n_bits,
-            fit_encrypted=True,
-            random_state=random_state,
-            parameters_range=parameters_range,
-            max_iter=max_iter,
-            warm_start=True,
-            early_stopping=False,
-            fit_intercept=fit_intercept,
-            verbose=True,
-        )
-        model_from_another.training_p_error = 1e-15
-
-    with warnings.catch_warnings():
-        warnings.filterwarnings(
-            "ignore",
-            message="ONNX Preprocess - Removing mutation from node aten::sub_ on block input.*",
-        )
+    # Define mixed fitting parameters
+    mixed_fit_init_kwargs = {"warm_start": True}
+
+    weights_mixed, bias_mixed, y_pred_proba_mixed, y_pred_class_mixed = check_encrypted_fit(
+        x,
+        y,
+        n_bits,
+        random_state,
+        parameters_range,
+        max_iter,
+        fit_intercept,
+        check_accuracy,
+        mixed_fit=True,
+        init_kwargs=mixed_fit_init_kwargs,
+    )
 
-        model_from_another.fit(
-            x,
-            y,
-            fhe="disable",
-            coef_init=model_partial.coef_,
-            intercept_init=model_partial.intercept_,
-        )
+    # Make sure weight, bias and prediction values are identical between clear and mixed fitting
+    assert array_allclose_and_same_shape(weights_disable, weights_mixed)
+    assert array_allclose_and_same_shape(bias_disable, bias_mixed)
+    assert array_allclose_and_same_shape(y_pred_proba_disable, y_pred_proba_mixed)
+    assert array_allclose_and_same_shape(y_pred_class_disable, y_pred_class_mixed)
+
+    # Define warn start parameters
+    warm_start_fit_kwargs = {
+        "coef_init": weights_disable,
+        "intercept_init": bias_disable,
+    }
+
+    # Because the model's `random_number_generator` will be re-initialized with the same random
```

Comment:
I tried to add checking values coherence at the end but unfortunately the fact that we re-initialize the rng instance at each model init with the exact same random_state more or less prevents us to do so

Reply:
just changed and made passing the RNG object between calls possible indeed, the test is green , thanks !